decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
- [ ] Please make sure that you specify the relationship between the Decorator class and the Nameable class to ensure the correctness of the inheritance.
attr_accessor :nameable